providers/providers_manager.go

```diff
@@ -317,6 +361,7 @@ func (pm *ProviderManager) getProviderSetForKey(ctx context.Context, k []byte) (
 	}
 
 	if len(pset.providers) > 0 {
+		// TODO: if this is a prefix, do we want to cache it?
```

Comment:
Is it possible to add only the target CID to the cache (if found), and not all content matching the prefix?

Reply:
yes, I can change this to only cache full CIDs!